### ISSUE TITLE
Exposing the sender in handle_info

### ISSUE THREAD

**Issue by easco** [BlakeWilliams/Elixir-Slack#108]
Because `handle_info` receives any messages sent to the server that are not handled elsewhere, it would be convenient channel to get information (e.g. about the slack connection) from the server.  However, because the process that sent the message is not exposed in handle_info, there is no way to send a message back to the server.  

If possible, it would be nice to include the sender in the parameters to handle_info so information could be returned.

**Comment by BlakeWilliams** [comment]
I think there's an example of using `handle_info` in the README, is that not correct?

**Comment by easco** [comment]
I've done you a disservice and not done well in my communication.  I tried to offer a proposed solution without explaining my problem.  Thank you for your patience and let me try again.

From what I understand, the Slack Bot runs as a `GenServer-like` process.  Primarily it sits in a receive loop and responds to messages from a web socket connection to Slack. 

One of the advantages of a `GenServer` is the fact that I can send it a message and get information back - get information out of the process.  What I want to do is send the Slack Bot process a message and get back information about the connection the bot has established with Slack.  For example, I'd like to have a separate process in my application send my Slack Bot a "whoami" message and get back the user name that the bot is connected as (so slack.me.name)

As it stands, I don't see any way to do that. All of the message handlers in the bot's process are concerned with information coming from the web socket. There're no handlers for message sent by other processes with a link back to the caller that I could use to return information.

The solution that came to mind would be to change `handle_info` so that,  like `handle_call` for a `GenServer`, one of the parameters is the process that sent the message.  That way I could send an arbitrary message (like `:whoami`) into the bot process and it could respond to the caller with the user name (e.g. `slack.me.name`) of the connection.

**Comment by BlakeWilliams** [comment]
Sorry for the delay on this. That sounds like a good idea.

I'd likely accept a PR that exposes that information.

**Comment by acconrad** [comment]
closing as over 3 years old
